### PR TITLE
Detect disable aderyn comments

### PR DESCRIPTION
**Pull request by TilakMaddy**
Context 
Prerequisite for https://github.com/Cyfrin/aderyn-vscode/issues/20

This PR brings in:
* Option 1
* Type 1 & 2

**Caveat**

For this to work, you can't have `--skip-cloc` option enabled

### CODE REVIEW

File: tests/contract-playground/src/ConstantsLiterals.sol [hunk 1]
```diff
@@ -29,6 +29,7 @@ contract ConstantsLiterals {
         address multipleUseOfAddress = 0x95222290DD7278Aa3Ddd389Cc1E1d165CC4BAfe5;
         multipleUseOfAddress = 0x95222290DD7278Aa3Ddd389Cc1E1d165CC4BAfe5;
         bytes32 multipleUseOfBytes32 = 0x8a1b3dbe6301650442bfa765d4de23775fc9a4ec4329ebb5995ec7f1e3777dc4;
+        // aderyn-disable-next-line
         multipleUseOfBytes32 = 0x8a1b3dbe6301650442bfa765d4de23775fc9a4ec4329ebb5995ec7f1e3777dc4;
```
Comment (TilakMaddy) [inline tests/contract-playground/src/ConstantsLiterals.sol:33]: Type 1

File: tests/contract-playground/src/uniswap/UniswapV2Swapper.sol [hunk 2]
```diff
@@ -25,7 +25,7 @@ contract UniswapV2Swapper {
         router1.swapExactETHForTokens(amountOutMin, path, to, block.timestamp);
         router1.swapTokensForExactETH(amountOut, amountInMax, path, to, block.timestamp);
         router1.swapExactTokensForETH(amountIn, amountOutMin, path, to, block.timestamp);
-        router1.swapETHForExactTokens(amountOut, path, to, block.timestamp);
+        router1.swapETHForExactTokens(amountOut, path, to, block.timestamp); // aderyn-disable-line
```
Comment (TilakMaddy) [inline tests/contract-playground/src/uniswap/UniswapV2Swapper.sol:28]: Type 2

File: reports/report.md [hunk 3]
```diff
@@ -192,12 +192,6 @@ In the PoS model, proposers know well in advance if they will propose one or con
 	        router1.swapExactTokensForETH(amountIn, amountOutMin, path, to, block.timestamp);
 	```
 
-- Found in src/uniswap/UniswapV2Swapper.sol [Line: 28](../tests/contract-playground/src/uniswap/UniswapV2Swapper.sol#L28)
-
-	```solidity
-	        router1.swapETHForExactTokens(amountOut, path, to, block.timestamp);
-	```
-
```
Comment (TilakMaddy) [inline reports/report.md:200]: Line targetted by Type 2 comment ignored

File: reports/report.md [hunk 4]
```diff
@@ -1433,12 +1427,6 @@ If the same constant literal value is used multiple times, create a constant sta
 	        bytes32 multipleUseOfBytes32 = 0x8a1b3dbe6301650442bfa765d4de23775fc9a4ec4329ebb5995ec7f1e3777dc4;
 	```
 
-- Found in src/ConstantsLiterals.sol [Line: 32](../tests/contract-playground/src/ConstantsLiterals.sol#L32)
-
-	```solidity
-	        multipleUseOfBytes32 = 0x8a1b3dbe6301650442bfa765d4de23775fc9a4ec4329ebb5995ec7f1e3777dc4;
-	```
-
```
Comment (TilakMaddy) [inline reports/report.md:1441]: Line targetted by Type 1 comment ignored

**Review by alexroan — COMMENTED** [review]
@TilakMaddy This looks good.

QQ: What was your rationale for processing the ignore statements in fscloc instead of using tools like `peek` and `peek_above`?

**Comment by TilakMaddy** [comment]
> @TilakMaddy This looks good.
> 
> QQ: What was your rationale for processing the ignore statements in fscloc instead of using tools like `peek` and `peek_above`?

While we can work it out using `peek`/ `peek_over`, implementing Type 3 comment [https://github.com/Cyfrin/aderyn-vscode/issues/20#issuecomment-2184084795] would mean we will have a lot of edge cases to deal with. 

The **underlying** reason I feel is that `peek` methods are meant to give us hints to describe/discover something about a `node` written in English. (Almost like they are properties of the node)

Whereas `disable-next-line` is **not** descriptive of the node itself. It's a user preference for aderyn. So we need to be comment centric instead of node centric while approaching this issue (IMO)

Here, I used `fscloc` as a hook because we already get access to the tokenized descriptions (meaning we only have to do `string.contains` on the *Comment* tokens for `aderyn-disable-*` , unlike `peek` methods (which have to retokenize the snippet of the node's code)

**Review by alexroan — CHANGES_REQUESTED** [review]
Sweet. @TilakMaddy we should aim to get this in ASAP, but with a few minor DevEx changes:

1. This should function regardless of whether or not the `--skip-cloc` flag is set.
2. `aderyn-ignore-next-line` and `aderyn-ignore` instead

Once these two are resolved, we're good to go I think! :)

File: reports/report.json [hunk 5]
```diff
@@ -286,29 +286,23 @@
...
+            "src_char": "1400:99"
           },
           {
             "contract_path": "src/uniswap/UniswapV2Swapper.sol",
             "line_no": 33,
-            "src": "1492:109",
-            "src_char": "1492:109"
+            "src": "1509:109",
+            "src_char": "1509:109"
           },
```
Comment (TilakMaddy) [inline reports/report.json:306]: WHy have these changed ?

Comment (TilakMaddy) [reply reports/report.json:306]: Ahh now I see, it's because I added the comments to ignore, which changes the offset of existing nodes. All cool !

**Comment by TilakMaddy** [comment]
> Sweet. @TilakMaddy we should aim to get this in ASAP, but with a few minor DevEx changes:
> 
> 1. This should function regardless of whether or not the `--skip-cloc` flag is set.
> 2. `aderyn-ignore-next-line` and `aderyn-ignore` instead
> 
> Once these two are resolved, we're good to go I think! :)

DONE ✅

**Review by alexroan — APPROVED** [review]

**Pull request merged**